### ISSUE TITLE
NetworkManager block not ubdating non primary connections

### ISSUE THREAD

**Issue by TilCreator** [greshake/i3status-rust#1117]
I have noticed that the NetworkManager block doesn't update when my wifi connection changes state while being connected to a lan cable.
After a bit of digging I found that the NM block only subscribes to `PropertiesChanged` of `/org/freedesktop/NetworkManager` and not to every single connection.
https://github.com/greshake/i3status-rust/blob/094cfdc13953b31edda8914f5e536e52b93deb38/src/blocks/networkmanager.rs#L553-L556
This detects _almost_ every change, but it doesn't detect when a connection changes that is not the primary connection, because no property of `/org/freedesktop/NetworkManager` is changed.

This is a problem when for example: The computer has two internet connections, one lan and one wifi, the lan connection is the primary. Now when the wifi connection is disabled and re-enabled it changes the `connections` property of `/org/freedesktop/NetworkManager`, but it will be marked as disconnected because the connection is not established jet. The event of the wifi connection being established fully never reaches the block, because no property of `/org/freedesktop/NetworkManager` changes as a consequence.

One fix for this would be to subscribe to `PropertiesChanged` of every connection, but that is not super easy and maybe there is a better solution anyways. Maybe someone who knows dbus better than me knows of some was to subscribe to `/org/freedesktop/NetworkManager/connections/*` with one match rule or something even easier.
If I don't find a better solution I will just implement this one (in a few days).

**Comment by TilCreator** [comment]
Which version of NM are you using? I'm using 1.30.0-2.

**Comment by GladOSkar** [comment]
`1.30.0`

**Comment by TilCreator** [comment]
Weird

**Comment by TilCreator** [comment]
How to reproduce (for me):
`config_test_nm.toml`:
```toml
[[block]]
block = "networkmanager"
max_ssid_width = 10
ap_format = "{ssid} {strength}%"
device_format = "{icon}{name} {ap}"
connection_format = "{devices}"
interface_name_exclude = ["lxdbr\\d+", "virbr\\d+"]
interface_name_include = ["wl.+", "enp.+", "wg\\d+"]
```
commented result (dupplicate updates are removed to reduce confusion):
```rust
$ i3status-rs config_test_nm.toml
{"version": 1, "click_events": true}
[[{"full_text":"|","color":"#a9a9a9","background":"#000000","separator":false,"separator_block_width":0,"markup":"pango"},{"full_text":" ETH enp3s0f0  ","color":"#859900","background":"#000000","name":"0","instance":"0","separator":false,"separator_block_width":0,"markup":"pango"}],

// activate wifi connection: `nmcli connection up Spielberg`
// wifi connection gets displayed with color `#b58900` which means state is `disconnected`.
[{"full_text":"|","color":"#a9a9a9","background":"#000000","separator":false,"separator_block_width":0,"markup":"pango"},{"full_text":" ETH enp3s0f0  ","color":"#859900","background":"#000000","name":"0","instance":"0","separator":false,"separator_block_width":0,"markup":"pango"},{"full_text":" WLAN wlp1s0 Spielberg 97% ","color":"#b58900","background":"#000000","name":"0","instance":"0","separator":false,"separator_block_width":0,"markup":"pango"}],

// activate another connection to trigger a `PropertiesChanged` event: `nmcli connection up wg0`
// wifi connection gets displayed with color `#859900` which means state is `connected`.
[{"full_text":"|","color":"#a9a9a9","background":"#000000","separator":false,"separator_block_width":0,"markup":"pango"},{"full_text":" ETH enp3s0f0  ","color":"#859900","background":"#000000","name":"0","instance":"0","separator":false,"separator_block_width":0,"markup":"pango"},{"full_text":" VPN wg0  ","color":"#859900","background":"#000000","name":"0","instance":"0","separator":false,"separator_block_width":0,"markup":"pango"},{"full_text":" WLAN wlp1s0 Spielberg 82% ","color":"#859900","background":"#000000","name":"0","instance":"0","separator":false,"separator_block_width":0,"markup":"pango"}],
```

**Comment by GladOSkar** [comment]
Oh sorry i'm stupid i was testing the `net` block instead of `networkmanager`. Sorry. I don't use the latter.

**Comment by TilCreator** [comment]
I was already wondering why you used `ifconfig` ^^

**Comment by ammgws** [comment]
I wonder if #991 is related?

Anyway one thing I don't understand is, if NetworkManager is managing all wired and wireless connections on your computer, then how can there be no change in the dbus properties? Or is it just the fact that the dbus interface we are monitoring is only for the primary connection, and the others are in another part of the tree?

Can you check using d-feet to see where the various properties are?


Dbus has some wildcard matching rules, perhaps they might be of use here? Search for arg0namespace and you should be able to find the section in the specification that mentions it.

**Comment by TilCreator** [comment]
#991 is likely related

**Comment by TilCreator** [comment]
The path where we need to listen for `PropertiesChanged` is `/org/freedesktop/NetworkManager/ActiveConnection/*`.
